test/jdk/sun/net/www/http/KeepAliveStream/KeepAliveStreamCloseWithWrongContentLength.java

```diff
@@ -23,75 +23,113 @@
+                // if the rouge client sends as least bytes as there is in getRequest1stLine it will be discarded and
+                // the test should proceed otherwise it should timeout on readNBytes below
+                do {
+                    if (clientSocket != null) {
+                        final String client = "%s:%d".formatted(
+                                clientSocket.getInetAddress().getHostAddress(),
+                                clientSocket.getPort()
+                        );
+                        try {
+                            clientSocket.close();
+                        }
+                        catch (IOException ioe) {
+                            ioe.printStackTrace();
+                        }
+                        finally {
+                            System.err.println("rogue client (%s) connection attempt, ignoring".formatted(client));
+                        }
+                    }
+                    clientSocket = serverSocket.accept();
+                    // read HTTP request from client
+                    clientBytes = new ByteArrayOutputStream();
+                    clientBytes.write(clientSocket.getInputStream().readNBytes(getRequest1stLine.getBytes().length));
                 }
-                OutputStreamWriter ow =
-                    new OutputStreamWriter((os = s.getOutputStream()));
-                ow.write("HTTP/1.0 200 OK\n");
+                while(!getRequest1stLine.equals(clientBytes.toString()));
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+            try (OutputStreamWriter outputStreamWriter = new OutputStreamWriter(clientSocket.getOutputStream())) {
+                outputStreamWriter.write("HTTP/1.0 200 OK\n");
 
                 // Note: The client expects 10 bytes.
-                ow.write("Content-Length: 10\n");
-                ow.write("Content-Type: text/html\n");
+                outputStreamWriter.write("Content-Length: 10\n");
+                outputStreamWriter.write("Content-Type: text/html\n");
 
                 // Note: If this line is missing, everything works fine.
-                ow.write("Connection: Keep-Alive\n");
-                ow.write("\n");
+                outputStreamWriter.write("Connection: Keep-Alive\n");
+                outputStreamWriter.write("\n");
 
                 // Note: The (buggy) server only sends 9 bytes.
-                ow.write("123456789");
-                ow.flush();
-            } catch (Exception e) {
-            } finally {
-                try {if (os != null) { os.close(); }} catch (IOException e) {}
+                outputStreamWriter.write("123456789");
+                outputStreamWriter.flush();
             }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        @Override
+        public void close() throws Exception {
+            clientSocket.close();
```

Comment:
you should really guard against null here (using a local variable to capture clientSocket) since clientSocket could be transiently null until the run() method is called.